x-pack/plugins/fleet/server/services/epm/package_service.ts

```diff
@@ -189,7 +202,7 @@ class PackageClientImpl implements PackageClient {
     packageInfo: InstallablePackage,
     assetPaths: string[]
   ): Promise<InstalledAssetType[]> {
-    await this.#runPreflight();
+    await this.#runPreflight(INSTALL_PACKAGES_AUTHZ);
```

Comment:
`reinstallEsAssets` is called by the `metadata-check-transforms-task` security task, verified locally that it still works, even when logging in with a ready only integrations user, as it is triggered by the system

```
tySolution.endpoint:metadata-check-transforms-task:0.0.2] missing endpoint transforms found, attempting reinstall
[2023-08-14T11:02:30.687+02:00][DEBUG][plugins.fleet] Started transform: endpoint.metadata_current-default-8.9.1
[2023-08-14T11:02:30.688+02:00][DEBUG][plugins.fleet] Started transform: endpoint.metadata_united-default-8.9.1
```